spec/lib/api/hub_config_api_spec.rb

```diff
@@ -0,0 +1,121 @@
+        .to_return(status: 200)
+
+      response = hub_config_api.healthcheck
+
+      expect(response.status).to eq(200)
+    end
+
+    it 'should return a non-200 status code when the API responds with an error' do
+      stub_request(:get, 'http://config-service.test:80/service-status')
+        .to_return(status: 502)
+
+      response = hub_config_api.healthcheck
+
+      expect(response.status).to eq(502)
+    end
+  end
+
+  describe '#encryption_certificate' do
+    it 'should return the encryption certificate' do
+      stub_request(:get, "http://config-service.test:80/config/certificates/#{CGI.escape(entity_id)}/certs/encryption")
+        .to_return(body: hub_response_for_encryption.to_json)
+
+      response = hub_config_api.encryption_certificate(entity_id)
+
+      expect(response).to eq(encryption_certificate)
+    end
+
+    it 'should return nil and log error if certificate not found' do
+      stub_request(:get, "http://config-service.test:80/config/certificates/#{CGI.escape(wrong_entity_id)}/certs/encryption")
+        .to_return(status: 404, body: "{\"code\":404,\"message\":\"'#{wrong_entity_id}' - No data is configured for this entity.\"}")
+      
+      expect(Rails.logger).to receive(:error).with("Error getting encryption certificate for entity_id: #{wrong_entity_id}! (Code: 404)")
+
+      response = hub_config_api.encryption_certificate(wrong_entity_id)
+
+      expect(response).to be nil
+    end
+  end
+
+  describe '#signing_certificates' do
+    it 'should return an array of signing certs with one cert when not dual-running' do
+      stub_request(:get, "http://config-service.test:80/config/certificates/#{CGI.escape(entity_id)}/certs/signing")
+        .to_return(body: hub_response_for_signing.to_json)
+
+      response = hub_config_api.signing_certificates(entity_id)
+
+      expect(response).to eq([signing_certificate_one])
+    end
+
+    it 'should return an array of signing certs with two certs when dual-running' do
+      stub_request(:get, "http://config-service.test:80/config/certificates/#{CGI.escape(entity_id)}/certs/signing")
+        .to_return(body: hub_response_for_signing_when_dual_running.to_json)
+
+      response = hub_config_api.signing_certificates(entity_id)
+
+      expect(response).to eq([signing_certificate_one, signing_certificate_two])
+    end
+
+    it 'should return an empty array and log error if wrong entity-id' do
+      stub_request(:get, "http://config-service.test:80/config/certificates/#{CGI.escape(wrong_entity_id)}/certs/signing")
+        .to_return(status: 404, body: "{\"code\":404,\"message\":\"'#{wrong_entity_id}' - No data is configured for this entity.\"}")
+
+      expect(Rails.logger).to receive(:error).with("Error getting signing certificates for entity_id: #{wrong_entity_id}! (Code: 404)")
+
+      response = hub_config_api.signing_certificates(wrong_entity_id)
+
+      expect(response).to eq([])
```

Comment:
why not `expect(response).to be_empty`

Reply:
because the method should return an empty array (`[]`) not empty response/string... the test would still pass if the method started returning an empty string (`''`) which would be wrong